### PR TITLE
vpc,network: fix createLoadBalancer access on user network

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

While checking network access for creating load-balancer use `AccessType.OperateEntry`
Refactor variable name in `NetworkModelImpl::checkNetworkPermissions`

Fixes: #6590

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test centos7 vmware-67u3

**Review by weizhouapache — APPROVED** [review]
code lgtm

it looks this issue has existed for many years. should we fix it in older versions as well ?

**Review by rohityadavcloud — APPROVED** [review]

File: server/src/main/java/com/cloud/network/NetworkModelImpl.java [hunk 1]
```diff
@@ -1659,7 +1659,7 @@ public void checkCapabilityForProvider(Set<Provider> providers, Service service,
     }
 
     @Override
-    public void checkNetworkPermissions(Account owner, Network network) {
+    public void checkNetworkPermissions(Account caller, Network network) {
```
Comment (rohityadavcloud) [inline server/src/main/java/com/cloud/network/NetworkModelImpl.java:1662]: could we explore if a unit test is written

Comment (shwstppr) [reply server/src/main/java/com/cloud/network/NetworkModelImpl.java:1662]: added some unit tests

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@weizhouapache do you think this should go into 4.16 or earlier?

**Comment by rohityadavcloud** [comment]
No I think, 4.17 + makes sense.

**Pull request merged**